gcloud-java-compute/src/test/java/com/google/gcloud/compute/StandardDiskConfigurationTest.java

```diff
@@ -0,0 +1,103 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+public class StandardDiskConfigurationTest {
+
+  private static final Long SIZE = 42L;
+  private static final DiskTypeId DISK_TYPE = DiskTypeId.of("project", "zone", "type");
+  private static final StandardDiskConfiguration DISK_CONFIGURATION =
+      StandardDiskConfiguration.builder()
+          .sizeGb(SIZE)
+          .diskType(DISK_TYPE)
+          .build();
+
+  @Test
+  public void testToBuilder() {
+    compareDefaultDiskConfiguration(DISK_CONFIGURATION, DISK_CONFIGURATION.toBuilder().build());
+    StandardDiskConfiguration diskConfiguration = DISK_CONFIGURATION.toBuilder()
+        .sizeGb(24L)
+        .build();
+    assertEquals(24L, diskConfiguration.sizeGb().longValue());
+    diskConfiguration = diskConfiguration.toBuilder()
+        .sizeGb(SIZE)
+        .build();
+    compareDefaultDiskConfiguration(DISK_CONFIGURATION, diskConfiguration);
+  }
+
+  @Test
+  public void testToBuilderIncomplete() {
+    StandardDiskConfiguration diskConfiguration = StandardDiskConfiguration.of(DISK_TYPE);
+    compareDefaultDiskConfiguration(diskConfiguration, diskConfiguration.toBuilder().build());
+  }
+
+  @Test
+  public void testBuilder() {
+    assertEquals(DISK_TYPE, DISK_CONFIGURATION.diskType());
+    assertEquals(SIZE, DISK_CONFIGURATION.sizeGb());
+    assertEquals(Type.STANDARD, DISK_CONFIGURATION.type());
+  }
+
+  @Test
+  public void testToAndFromPb() {
+    assertTrue(DiskConfiguration.fromPb(DISK_CONFIGURATION.toPb())
+        instanceof StandardDiskConfiguration);
+    compareDefaultDiskConfiguration(DISK_CONFIGURATION,
+        DiskConfiguration.<StandardDiskConfiguration>fromPb(DISK_CONFIGURATION.toPb()));
+  }
+
+  @Test
+  public void testOf() {
+    StandardDiskConfiguration configuration = StandardDiskConfiguration.of(DISK_TYPE);
+    assertEquals(DISK_TYPE, configuration.diskType());
+    assertNull(configuration.sizeGb());
+    assertEquals(Type.STANDARD, configuration.type());
+    configuration = StandardDiskConfiguration.of(DISK_TYPE, SIZE);
+    assertEquals(DISK_TYPE, configuration.diskType());
+    assertEquals(SIZE, configuration.sizeGb());
+    assertEquals(Type.STANDARD, configuration.type());
+    configuration = StandardDiskConfiguration.of(SIZE);
+    assertNull(configuration.diskType());
+    assertEquals(SIZE, configuration.sizeGb());
+    assertEquals(Type.STANDARD, configuration.type());
+  }
+
+  @Test
+  public void testSetProjectId() {
+    StandardDiskConfiguration configuration = DISK_CONFIGURATION.toBuilder()
+        .diskType(DiskTypeId.of(DISK_TYPE.zone(), DISK_TYPE.diskType()))
+        .build();
+    compareDefaultDiskConfiguration(DISK_CONFIGURATION, configuration.setProjectId("project"));
+  }
+
+  private void compareDefaultDiskConfiguration(StandardDiskConfiguration expected,
```

Comment:
Nit: could call this `compareStandardDiskConfiguration` (to align with the class name)